src/pybamm/input/parameters/base_parameter_set.py

```diff
@@ -0,0 +1,39 @@
+class AbstractBaseParameters:
```

Comment:
Why is it AbstractBaseParameters and not just BaseParameters? If anything, these would be battery parameters (rather than abstract parameters). Also, might be worth to use BaseParameterSet or BaseParameterValues instead. I think we are not 100% consistent with this, but in theory parameters are the concept (e.g. diffusion coefficient) while parameter values/sets are the value.

Reply:
Abstract as in an abstract class where implementation details are not fully defined. The idea was that this provides an interface that we can extend for the various parameter sets.

Reply:
+1 for changing to `AbstractBaseParameterSets`